src/agent/main.cpp

```diff
@@ -107,17 +107,17 @@ static void HandleSignal(int aSignal)
 
 static int Mainloop(otbr::AgentInstance &aInstance, const char *aInterfaceName)
 {
-    int error = EXIT_FAILURE;
+    int                   error         = EXIT_FAILURE;
+    ControllerOpenThread *ncpOpenThread = reinterpret_cast<ControllerOpenThread *>(&aInstance.GetNcp());
```

Comment:
I lost the background of using reinterpret_cast here, but does static_cast work in this case?
```suggestion
    ControllerOpenThread &ncpOpenThread = reinterpret_cast<ControllerOpenThread &>(aInstance.GetNcp());
```

Reply:
fixed